### DISCUSSION TITLE
Учебный курс: Как заполнить строчную часть передав данные в хедер?

### DISCUSSION THREAD

**Discussion by d-nikolaev** [totumonline/totum-ru-issues-and-discussions#155] [Неопределенное]
По этому [уроку](https://ru.docs.totum.online/how-do-I-fill-in-the-line-part-by-passing-data-to-the-header) хочу создать отчёт.
Вроде всё сделал как надо но в h_insert не попадают данные. Подскажите данный пример из урока работает с расчётными таблицами?
Схема взаимодействия у меня построена так.
Расчётные таблицы.
![otchet](https://user-images.githubusercontent.com/13610260/172564231-d30f605f-4fc8-47c1-80c6-8e55ee31568d.png)
Кнопка расположена здесь:
![otchet_2](https://user-images.githubusercontent.com/13610260/172564385-d3ef16bc-c91f-4b8b-b218-b52af45b1432.png)
Получаю вот такой результат
![otchet_3](https://user-images.githubusercontent.com/13610260/172564434-6b00139d-3502-4cee-aa15-888cce83b18d.png)
Код действия:
![otchet_4](https://user-images.githubusercontent.com/13610260/172564785-177edef3-a691-4951-b68e-0a935bc35bcc.png)

**Comment by ttm-alexeypolunin** [comment]
Я вижу, что в строке `val` — `selectRowList` вроде как прям из таблицы внутри цикла (и первый скриншот вроде тоже про это говорит), но параметра `cycle` не указано.

Скорее всего здесь логически по архитектуре нечетко. **Что должно быть в выведено в Отчет?**

**Reply by d-nikolaev** [reply]
Вот эти данные пытаюсь получить в отчёт
![2022-06-08_114103](https://user-images.githubusercontent.com/13610260/172572906-997aafe2-9bef-4541-8dd8-74587d7c8c72.png)

**Reply by d-nikolaev** [reply]
Мне кажется я понял как надо поступить. 
![2022-06-08_115044](https://user-images.githubusercontent.com/13610260/172574909-27da9a2a-759b-41ab-b940-8a8eb19e3e88.png)
Из цикла вывести данные в табличную часть расчётной таблицы а потом передать их в отчёт.

**Reply by ttm-alexeypolunin** [reply]
угу

**Reply by d-nikolaev** [reply]
Можно я задам вопрос в этой ветке?
Он не касается данной темы но касается обсуждаемой таблицы.
Через ярлык я получил данные
![gd-1](https://user-images.githubusercontent.com/13610260/173510835-e7d246d7-7770-4574-8fad-4ffee46b4c4a.png)
Хочу изменить формат данных только год
![gd-2](https://user-images.githubusercontent.com/13610260/173510900-48c283fb-9b55-4818-a055-5274d6a22f59.png)
Не работает. Видимо не там (не в то место) ввожу функцию. 

P.S: Уверен в учебном курсе есть ответ на этот вопрос, но пока не дошел видимо до этого места, а нужно это сделать сейчас чтобы вести разработку дальше.

**Reply by d-nikolaev** [reply]
Или я внутри цикла создаю дополнительное поле в котором делаю формат по типу год, а уже потом вывожу как дополнительное поле в табличную часть расчётной таблицы?

**Reply by ttm-alexeypolunin** [reply]
Ярлык берет настройки поля источника.

Если вы хотите форматом переписать отображение даты, то в секцию `f` в поле должен приходить результат работы функции `setFormat`.

```
f1=: setFormat(text: $date)

date: dateFormat(date: $#ntn; format: "Y")
```

Если вы хотите, что бы это значение бралось, когда пользователь копирует значение поля — добавить надо `textasvalue`: 

```
f1=: setFormat(text: $date; textasvalue: true)

date: dateFormat(date: $#ntn; format: "Y")
```

**Comment by d-nikolaev** [comment]
Вроде как сделал
![gd-3](https://user-images.githubusercontent.com/13610260/173537860-9019bbc8-72b4-4637-b4fe-70104487187c.png)
Но не до конца уверен что правильно. Вдруг неверные подходы делал. =)
